// csp-header/dist/constants/values.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NO_REFERER = exports.INLINE = exports.HASHES = exports.EVAL = exports.UNSAFE_URL = exports.UNSAFE_INLINE = exports.UNSAFE_HASHES = exports.UNSAFE_EVAL = exports.STRICT_DYNAMIC = exports.SELF = exports.SCRIPT = exports.ORIGIN_WHEN_CROSS_ORIGIN = exports.ORIGIN = exports.NONE_WHEN_DOWNGRADE = exports.NONE = exports.NO_REFERRER = exports.DATA = exports.BLOB = exports.ALLOW_TOP_NAVIGATION_BY_USER_ACTIVATION = exports.ALLOW_TOP_NAVIGATION = exports.ALLOW_SCRIPTS = exports.ALLOW_STORAGE_ACCESS_BY_USER_ACTIVATION = exports.ALLOW_SAME_ORIGIN = exports.ALLOW_PRESENTATION = exports.ALLOW_POPUPS_TO_ESACPE_SANDBOX = exports.ALLOW_POPUPS = exports.ALLOW_POINTER_LOCK = exports.ALLOW_ORIENTATION_LOCK = exports.ALLOW_MODALS = exports.ALLOW_FORMS = exports.ALLOW_DUPLICATES = exports.ALLOW_DOWNLOADS_WITHOUT_USER_ACTIVATION = void 0;
exports.ALLOW_DOWNLOADS_WITHOUT_USER_ACTIVATION = "allow-downloads-without-user-activation";
exports.ALLOW_DUPLICATES = "'allow-duplicates'";
exports.ALLOW_FORMS = "allow-forms";
exports.ALLOW_MODALS = "allow-modals";
exports.ALLOW_ORIENTATION_LOCK = "allow-orientation-lock";
exports.ALLOW_POINTER_LOCK = "allow-pointer-lock";
exports.ALLOW_POPUPS = "allow-popups";
exports.ALLOW_POPUPS_TO_ESACPE_SANDBOX = "allow-popups-to-escape-sandbox";
exports.ALLOW_PRESENTATION = "allow-presentation";
exports.ALLOW_SAME_ORIGIN = "allow-same-origin";
exports.ALLOW_STORAGE_ACCESS_BY_USER_ACTIVATION = "allow-storage-access-by-user-activation";
exports.ALLOW_SCRIPTS = "allow-allow-scripts";
exports.ALLOW_TOP_NAVIGATION = "allow-top-navigation";
exports.ALLOW_TOP_NAVIGATION_BY_USER_ACTIVATION = "allow-top-navigation-by-user-activation";
exports.BLOB = 'blob:';
exports.DATA = 'data:';
exports.NO_REFERRER = "'no-referrer'";
exports.NONE = "'none'";
exports.NONE_WHEN_DOWNGRADE = "'none-when-downgrade'";
exports.ORIGIN = "'origin'";
exports.ORIGIN_WHEN_CROSS_ORIGIN = "'origin-when-cross-origin'";
exports.SCRIPT = "'script'";
exports.SELF = "'self'";
exports.STRICT_DYNAMIC = "'strict-dynamic'";
exports.UNSAFE_EVAL = "'unsafe-eval'";
exports.UNSAFE_HASHES = "'unsafe-hashes'";
exports.UNSAFE_INLINE = "'unsafe-inline'";
exports.UNSAFE_URL = "'unsafe-url'";
// @deprecated Use UNSAFE_EVAL
exports.EVAL = "'unsafe-eval'";
// @deprecated Use UNSAFE_HASHES
exports.HASHES = "'unsafe-hashes'";
// @deprecated Use UNSAFE_INLINE
exports.INLINE = "'unsafe-inline'";
// @deprecated Use NO_REFERRER
exports.NO_REFERER = exports.NO_REFERRER;
